enumerables.rb

```diff
@@ -0,0 +1,153 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    while i < to_a.length
+      yield to_a[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    result = []
+    to_a.my_each { |item| result << item if yield(item) }
+    result
+  end
+
+  def my_all?
+    to_a.my_each do |i|
+      return false if yield(i) == false
+      else
+        return true
+      end
+    end
+  end
+
+  def my_any?
+    return to_enum(:my_any?) unless block_given?
+
+    i = 0
+    ans = false
+    to_a.length.times do
+      ans = yield to_a[i]
+
+      break if ans == true
+
+      i += 1
+    end
+    ans
+  end
```

Comment:
***my_any?***

I will be using these variables for the next suggestions:

```
true_array = [1, true, 'hi', []]
array = [2, 5, 9, 7]
```

- [ ] When no block or argument is given returns true if at least one of the collection is not false or nil

```
      true_array.my_any?
      
        Expected: true

        Got: #<Enumerator:47263469722140> => #<Enumerator: [nil, false, true, []]:my_any?>
```

- [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric (Your method should be able to take an argument, such as a Regex, a Class, or a number):

```
      array.my_any?(Numeric)
      
      ArgumentError:
        wrong number of arguments (given 1, expected 0)
      # ./enumerables.rb:39:in `my_any?'
```